core/commands/storage/upload/upload/filelist.go

```diff
@@ -0,0 +1,109 @@
+package upload
+
+import (
+	"context"
+	"time"
+
+	cmds "github.com/TRON-US/go-btfs-cmds"
+	config "github.com/TRON-US/go-btfs-config"
+	uh "github.com/TRON-US/go-btfs/core/commands/storage/upload/helper"
+	"github.com/tron-us/go-btfs-common/crypto"
+	guardpb "github.com/tron-us/go-btfs-common/protos/guard"
+	"github.com/tron-us/go-btfs-common/utils/grpc"
+)
+
+const (
+	renterFilePageSize = 100
+	guardTimeout       = 360 * time.Second
+)
+
+var FilelistCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "List file store information for renter.",
+		ShortDescription: `
+The command helps renter to retrieve the file information for storage, includes file hash, 
+file size, rent start and rent end. Especially for renewing the contract for a new extended 
+duration, renter can check files with the command and find the desired file to renew. 
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("file-hash", true, false, "Hash of file to renew."),
+		cmds.StringArg("renew-length", true, false, "New File storage period on hosts in days."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctxParams, err := uh.ExtractContextParams(req, env)
+		if err != nil {
+			return err
+		}
+		renterId := ctxParams.N.Identity.String()
+		var files []*guardpb.FileStoreMeta
+		for i := 0; ; i++ {
+			metaReq := &guardpb.ListRenterFileInfoRequest{
+				RenterPid:        renterId,
+				RequesterPid:     renterId,
+				RequestTime:      time.Now().UTC(),
```

Comment:
Should use the latest time from local cache.

Reply:
Maybe take a look at how `/storage/contracts/list` and `/storage/contracts/sync` are implemented. This way we do not need to call guard on every list. Most of the times can read from local cache, while background sync will refresh automatically.